akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/sse/EventStreamUnmarshalling.scala

```diff
@@ -29,26 +32,38 @@ object EventStreamUnmarshalling extends EventStreamUnmarshalling
  * Mixing in this trait lets a `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of
  * [[ServerSentEvent]]s.
  *
- * The maximum size for parsing server-sent events is 8KiB by default and can be customized by overriding
- * [[EventStreamUnmarshalling.maxEventSize]]. The maximum size for parsing lines of a server-sent event is 4KiB dy
- * default and can be customized by overriding [[EventStreamUnmarshalling.maxLineSize]].
+ * The maximum size for parsing server-sent events is 8KiB by default and can be customized by configuring
+ * `akka.http.sse.max-event-size`. The maximum size for parsing lines of a server-sent event is 4KiB by
+ * default and can be customized by configuring `akka.http.sse.max-line-size`.
  */
 @ApiMayChange
 trait EventStreamUnmarshalling {
 
+  /**
+   * The maximum size for parsing lines of a server-sent event; 4KiB by default.
+   */
+  @deprecated("Set this property in configuration as `akka.http.sse.max-line-size` before calling fromEventsStream(implicit ActorSystem)", "10.0.15")
+  protected def maxLineSize: Int = 4096
+
   /**
    * The maximum size for parsing server-sent events; 8KiB by default.
    */
-  protected def maxEventSize: Int =
-    8192
+  @deprecated("Set this property in configuration as `akka.http.sse.max-event-size` before calling fromEventsStream(implicit ActorSystem)", "10.0.15")
+  protected def maxEventSize: Int = 8192
+
+  @deprecated("Binary compatibility method. Invocations should have an implicit ActorSystem in scope to provide access to configuration", "10.0.15")
+  final val fromEventStream: FromEntityUnmarshaller[Source[ServerSentEvent, NotUsed]] =
+    fromEventsStream(maxEventSize, maxLineSize)
 
   /**
-   * The maximum size for parsing lines of a server-sent event; 4KiB by default.
+   * Lets an `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of `ServerSentEvent`s.
    */
-  protected def maxLineSize: Int =
-    4096
+  implicit final def fromEventsStream(implicit system: ActorSystem): FromEntityUnmarshaller[Source[ServerSentEvent, NotUsed]] = {
```

Comment:
Would it make sense to also provide a non-`implicit` method where you can bring your own `ServerSentEventSettings`? So (in rare cases) you could programmatically set different settings for different routes.